tests/Containers/SmallVectorTest.cpp

```diff
@@ -0,0 +1,216 @@
+
+  for (uint32_t i = 0; i < 1000; ++i) {
+    EXPECT_EQ(values.size(), i);
+    values.push_back(i);
+    EXPECT_EQ(values.size(), i + 1);
+  }
+  values.clear();
+  for (uint32_t i = 0; i < 512; ++i) {
+    values.emplace_back(i * 2);
+  }
+  EXPECT_EQ(values.size(), 512);
+  for (uint32_t i = 0; i < 512; ++i) {
+    EXPECT_EQ(i * 2, values[i]);
+    EXPECT_EQ(i * 2, values.at(i));
+  }
+}
+
+TEST(SmallVectorTest, test_resize) {
+  arangodb::containers::SmallVectorWithArena<uint32_t, 32> values;
+
+  values.resize(10000);
+  EXPECT_EQ(values.size(), 10000);
+  for (uint32_t i = 0; i < 10000; ++i) {
+    EXPECT_EQ(0, values[i]);
+    EXPECT_EQ(0, values.at(i));
+  }
+
+  values.emplace_back(1);
+  EXPECT_EQ(values.size(), 10001);
+  EXPECT_EQ(1, values[10000]);
+  EXPECT_EQ(1, values.at(10000));
+
+  values.resize(3);
+  EXPECT_EQ(values.size(), 3);
+  for (uint32_t i = 0; i < 3; ++i) {
+    EXPECT_EQ(0, values[i]);
+    EXPECT_EQ(0, values.at(i));
+  }
+
+  values.resize(0);
+  EXPECT_EQ(values.size(), 0);
+}
+
+TEST(SmallVectorTest, test_at) {
+  arangodb::containers::SmallVectorWithArena<uint32_t, 32> values;
+
+  EXPECT_THROW(values.at(0), std::out_of_range);
+  EXPECT_THROW(values.at(1), std::out_of_range);
+  EXPECT_THROW(values.at(12345), std::out_of_range);
+
+  values.resize(10);
+  EXPECT_EQ(0, values.at(9));
+  EXPECT_THROW(values.at(10), std::out_of_range);
+}
+
+TEST(SmallVectorTest, test_front) {
+  arangodb::containers::SmallVectorWithArena<uint64_t, 32> values;
+
+  values.push_back(666);
+  EXPECT_EQ(666, values.front());
+  
+  values.push_back(999);
+  EXPECT_EQ(666, values.front());
+
+  for (uint64_t i = 0; i < 10; ++i) {
+    values.push_back(i);
+    EXPECT_EQ(666, values.front());
+  }
+}
+
+TEST(SmallVectorTest, test_back) {
+  arangodb::containers::SmallVectorWithArena<uint64_t, 32> values;
+
+  values.push_back(666);
+  EXPECT_EQ(666, values.back());
+  
+  values.push_back(999);
+  EXPECT_EQ(999, values.back());
+
+  for (uint64_t i = 0; i < 10; ++i) {
+    values.push_back(i);
+    EXPECT_EQ(i, values.back());
+  }
+}
+
+TEST(SmallVectorTest, test_iterator) {
+  arangodb::containers::SmallVectorWithArena<uint64_t, 32> values;
+
+  EXPECT_EQ(values.begin(), values.end());
+
+  for (uint64_t i = 0; i < 100; ++i) {
+    values.push_back(i * 3);
+    EXPECT_NE(values.begin(), values.end());
+  }
+
+  uint64_t i = 0;
+  auto it = values.begin();
+  while (it != values.end()) {
+    EXPECT_EQ(i * 3, *it);
+    ++it;
+    ++i;
+  }
+}
+
+TEST(SmallVectorTest, test_reverse_iterator) {
+  arangodb::containers::SmallVectorWithArena<uint64_t, 32> values;
+
+  EXPECT_EQ(values.begin(), values.end());
+
+  for (uint64_t i = 0; i < 100; ++i) {
+    values.push_back(i * 3);
+    EXPECT_NE(values.begin(), values.end());
+  }
+
+  uint64_t i = 0;
+  auto it = values.rbegin();
+  while (it != values.rend()) {
+    EXPECT_EQ((100 - i - 1) * 3, *it);
+    ++it;
+    ++i;
+  }
+}
```

Comment:
I would like to have some tests that check that the initial capacity matches the specified buffer size and that the elements are actually stored inside the arena. I.e., insert capacity number of elements, and for each one check that (pseudocode) `&vec[i] >= &vec && &vec[i] <= &vec + sizeof(vec)`.

Reply:
Added an extra test.